src/BinarySerializationTestForStructs.zig

```diff
@@ -0,0 +1,137 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const file = std.fs.File;
+const cwd = std.fs.cwd();
+const fba = std.heap.FixedBufferAllocator;
+
+const ArbitraryType = struct {
+    size: u64,
+    line: []u8
+};
+const BoxedStructure = packed struct {
+    flag: bool,
+    letter: u8,
+    u32_value: u32,
+};
+const arbitrary_output_name: []const u8 = "./output/arbitrary.bin";
+const boxed_type_one_output_name: []const u8 = "./output/boxed_type_one.bin";
+
+pub fn WriteArbitraryType(mem_alloc: std.mem.Allocator) !ArbitraryType {
+    var at: ArbitraryType = .{
+        .size = 25,
+        .line = try mem_alloc.alloc(u8, 25)
+    };
+    // Zero out the u8 array to ensure no trash values
+    // are written to the file.
+    @memset(at.line, 0);
+    const hello_world = "Hello, World!";
+    var ii: u64 = 0;
+    while (ii < hello_world.len) : (ii += 1) {
+        at.line[ii] = hello_world[ii];
+    }
+    const arb_file = try cwd.createFile(arbitrary_output_name, .{.lock = .exclusive});
+    defer arb_file.close();
+    try arb_file.writer().writeAll(std.mem.asBytes(&at.size));
+    try arb_file.writer().writeAll(at.line);
+    return at;
+}
+pub fn ReadArbitraryType(mem_alloc: std.mem.Allocator) !ArbitraryType {
+    const arb_file = try cwd.openFile(arbitrary_output_name,
+        .{.mode = .read_only, .lock = .exclusive});
+    defer arb_file.close();
+    const last_pos = try arb_file.getEndPos();
+    const arbitrary_buffer: []u8 = try mem_alloc.alloc(u8, last_pos);
```

Comment:
It's a good idea to use `errdefer` to free memory if you're not sure the function will complete, which you're not since you use `try`s later on:
```zig
const arbitrary_buffer: []u8 = try mem_alloc.alloc(u8, last_pos);
errdefer mem_alloc.free(arbitrary_buffer);
```
See https://ziglang.org/documentation/master/#errdefer for more information.